### DISCUSSION TITLE
Official CHDMAN GUI

### DISCUSSION THREAD

**Discussion by RinMaru** [mamedev/discussions#89] [Ideas]
I'm not tech savvy and I imagine a good portion of users aren't. Chdman has become a golden standard for Roms and I think an official GUI for CHDMAN would be absolutely brilliant since it seems all other alternatives are out of date, and you can't beat support from the main devs.  Would this be possible or are there other means by which people can use Chdman?

I use CHDMAN for converting ISO, GDI, and CUE BINs to CHD and it's nice to have a Gui that can allow you to set a game as parent and child along with how many threads you want to use and the compression you want.

I really can't explain how maddening it is doing this in cmd so I'd really love a GUI or Frontend to allow me to visually see what I'm doing. all while exposing setting I can look into.

**Comment by sairuk** [comment]
imo a gui is completely unnecessary the tool functions fine for my purposes and is easily scriptable because it is not a gui app.

Maybe one day someone might feel like working on one or this might be a good entry point into the project for someone..

All this time spent creating discussions, polls and jumping on irc asking for someone to do this would arguably be better spent learning how to do this yourself and either run it as separate project or submit a pull request.

**Reply by RinMaru** [reply]
EDIT- sorry early morning and didn't read it completely.  Glad it's easy for your needs. I on the other hand have a difficult time inputting anything in cmd its honestly a real pita. using a GUI just makes things allot smoother for me and aware of what I'm doing. Love to see something like an interface that doesn't require studying an hour to finally know what i want to do with the vast amounts of options. Its not something the devs should worry about but I think it would be nice.

**Reply by sairuk** [reply]
OK i'd point out the irony here but you already missed my main point so good luck with your chase.

**Reply by cuavas** [reply]
It would lend itself to a “wizard” interface though.  You could step through choosing the task, selecting files, setting options, etc. and check that the settings are consistent as you go.

You wouldn’t even necessarily have to integrate the GUI with chdman itself, although you’d want to at least be able to read CHD headers to ensure that the options will work with the selected output parent CHD.

**Reply by sairuk** [reply]
While my opinion wont change for my particular use case nor others who I know run chdman headless, I appreciate there are folks out there that want gui and agree a wizard style approach might serve them well. I may be completely off the mark, but I'd see it as a better use of time for those with the skills to build chdman into an official library/spec that other tooling can incorporate (i.e. frontends, emus). Including having the standard documented and spec'd properly. iirc the doco was started, not sure where its at and one could argue source is doco but I digress.

I've always seen a frontend as a fairly entrypoint type experiment having written a few of my own over the years and wouldn't rate myself. As per op comments chd seems to have become the defacto gold standard for better or worse, couldn't say. I had/have every intent to incorporate a frontended chdman/floptool into retronas for our users at some point in time, both tui/web but it never entered my mind that this should be something the main project is managing, it is just down to me to finishing off our imager eventually and understanding a lot more about particular formats.

I do like the point about handling headers, that would be nice to get built into chdman for external use, so its one codebase

more than happy to be told i'm looking at this all wrong however.

**Reply by cuavas** [reply]
The chdman code is pretty dire to be honest, so turning that into a library might not be the greatest idea.

Fortunately, the bulk of the CHD handling code is already in lib/util and as such is isolated pretty well from the rest of MAME, and hence relatively easy to integrate into other projects.

The only parts of chdman you’d actually need for integrating the functionality into another tool (or into MAME’s internal UI for that matter) are the compressor classes.  But the whole interface for CHD creation should probably be reworked before that happens anyway.

I do agree that a front-end for chdman is probably something that could be handled well by someone outside the team.
